### ISSUE TITLE
Initial notification ignores custom notificationHandlerFactory

### ISSUE THREAD

**Issue by PhilLab** [gotev/android-upload-service#634]
**Describe the bug**
 The initial notification does not use the `NotificationHandlerFactory` which you can set by `UploadServiceConfig.setNotificationHandlerFactory()`. Instead it is created standalone:
 https://github.com/gotev/android-upload-service/blob/master/uploadservice/src/main/java/net/gotev/uploadservice/UploadService.kt#L206
 
 ```
         val notification = NotificationCompat.Builder(this, UploadServiceConfig.defaultNotificationChannel!!)
             .setSmallIcon(android.R.drawable.ic_menu_upload)
             .setOngoing(true)
             .setGroup(UploadServiceConfig.namespace)
             .build()
 
         startForeground(UPLOAD_NOTIFICATION_BASE_ID, notification)
 ```
 This means that if you applied any special handling, it won't work for the very first notification
 
 This is a bigger issue on Samsung devices, because they have a crash bug for which the workaround cannot be injected into the upload service:
 - Upload service crashes with exception `android.app.ForegroundServiceStartNotAllowedException: startForegroundService() not allowed due to mAllowStartForeground false: service com.example.app/net.gotev.uploadservice.UploadService`
 - Caused by a since-fixed Android 12 bug https://android-review.googlesource.com/c/platform/frameworks/support/+/2175727
 - But remains on Samsung devices: https://issuetracker.google.com/issues/229000935
 - Workaround: Add `if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.S) {    builder.setForegroundServiceBehavior(Notification.FOREGROUND_SERVICE_IMMEDIATE); }` when creating a notification in the background.
 
 **To Reproduce**
 1. Use the upload library on a Samsung device with Android 12 or higher
 2. Start an upload
 3. Eventually, this will result in a silent crash of your app (sometimes, it does not crash on the first try)
 
 **Expected behavior**
 The creation of the very first notification should also use the notification factory, if a custom one was provided.

**Comment by PhilLab** [comment]
I know, from #615 that the scenario the workaround aims to fix, is not necessarily supported. But from the factory perspective only, it could still be fixed.

**Comment by PhilLab** [comment]
Bump

**Comment by PhilLab** [comment]
@gotev would you consider this being relevant for the library?

**Comment by Gauravintelloger** [comment]
@gotev  can u  fixed  this issue?

**Comment by gotev** [comment]
@PhilLab thank you for this detailed report and proposed solution, much appreciated! Made a PR #648 

The very first notification is inteded to be constructed like that, as it serves the sole purpose of guaranteeing service launch. 

The NotificationHandler Factory serves a different purpose: to allow you to control how each of the uploads notifications are shown. The very first notification is something which Android requires to be there for the service to be launched. The NotificationHandler does its job after the service is started.

If the docs are not clear about this aspect, every suggestion is welcome to improve them.

**Comment by gotev** [comment]
Merged the PR and released version `4.9.2`
